### ISSUE TITLE
Backpropagation et loss function

### ISSUE THREAD

**Issue by Jefidev** [Jefidev/AIChess#3]
On sait que le réseau n'est backpropagé qu'une fois la partie finie. Néanmoins on ignore encore comment faire cela.

Commentez donc avec des ressources ou des explications quant à la **Backpropagation** et/ou à la fonction de **loss** à utiliser

**Comment by FrancoisGe** [comment]
Pour la backpropagation, on va comparer les prédictions du modèle (Value Actual State, (Proba Coup 1, PC2, ...) par rapport aux valeurs données par le MCTS (Value State) et pour la proba de choisir un coup, on reprend la liste donnée par le modèle, on retire les coups impossibles et on renormalise pour que la somme des probas restantes soit égale à 1. 

Modèle Output : Prédictions : float(VAS) , list(PC)
MCTS : True Value : float(VAS) ,
sum (normalise(predList(PC) * Vecteur des coups possibles [0,1,...,0] )==1

L'objectif est que le modèle apprenne les valeurs données par MCTS et apprenne les coups jouables ou non.
